### PR TITLE
Peress/fff migration ctrl isoal

### PR DESCRIPTION
**Pull request by yperess**

### CODE REVIEW

**Review by aaronemassey — DISMISSED** [review]

**Comment by cvinayak** [comment]
@wopu-ot please assign a reviewer to this PR as needed by your team

File: tests/bluetooth/ctrl_isoal/src/main.c [hunk 1]
```diff
@@ -375,30 +375,26 @@ static isoal_status_t custom_sink_sdu_alloc_test(const struct isoal_sink *sink_c
...
  */
-static isoal_status_t sink_sdu_emit_test(const struct isoal_sink *sink_ctx,
-					 const struct isoal_sdu_produced *valid_sdu)
+static isoal_status_t custom_sink_sdu_emit_test(const struct isoal_sink *sink_ctx,
+						const struct isoal_sdu_produced *valid_sdu)
 {
 	debug_trace_func_call(__func__, "IN");
 
 	debug_print_rx_sdu(sink_ctx, ((struct rx_sdu_frag_buffer *)valid_sdu->contents.dbuf)->sdu);
+	sink_sdu_emit_test_handler(*sink_ctx, *valid_sdu);
```
Comment (keith-zephyr) [inline tests/bluetooth/ctrl_isoal/src/main.c:394]: Hasn't this already been called by the isoal framework?

File: tests/bluetooth/ctrl_isoal/src/main.c [hunk 2]
```diff
@@ -328,6 +331,24 @@ static uint16_t add_framed_pdu_end(uint8_t *dataptr, uint8_t length, struct isoa
 	return insert_segment(true, true, 0, dataptr, length, pdu_meta);
 }
 
+FAKE_VALUE_FUNC(isoal_status_t, sink_sdu_alloc_test, const struct isoal_sink *,
+		const struct isoal_pdu_rx *, struct isoal_sdu_buffer *);
+
+static struct {
+	struct isoal_sdu_buffer *out[5];
```
[the remark this answers is not in the document]
Comment (yperess) [reply tests/bluetooth/ctrl_isoal/src/main.c:337]: It's the maximum number of times that we'll call `push_custom_sink_sdu_alloc_test_output_buffer` in any test. If a new test is added which exceeds this, it'll fail below on line 348 asserting that the size didn't overflow.

File: tests/bluetooth/ctrl_isoal/src/main.c [hunk 3]
```diff
@@ -955,6 +978,9 @@ ZTEST(test_rx_unframed, test_rx_unframed_single_pdu)
 		"FSM state %s should be %s!",
 		FSM_TO_STR(isoal_global.sink_state[sink_hdl].sdu_production.fsm),
 		FSM_TO_STR(ISOAL_START));
+	zassert_equal_ptr(&isoal_global.sink_state[sink_hdl], sink_sdu_alloc_test_fake.arg0_val,
+			  NULL);
+	zassert_equal_ptr(&rx_pdu_meta_buf.pdu_meta, sink_sdu_alloc_test_fake.arg1_val, NULL);
```
[the remark this answers is not in the document]
Comment (yperess) [reply tests/bluetooth/ctrl_isoal/src/main.c:921]: Done

**Review by aaronemassey — APPROVED** [review]

**Comment by wopu-ot** [comment]
@niag-Oticon please have a look

File: tests/bluetooth/ctrl_isoal/src/main.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (keith-zephyr) [reply tests/bluetooth/ctrl_isoal/src/main.c:927]: I don't  see a check of the `sdu_buffer` parameter.

File: tests/bluetooth/ctrl_isoal/src/main.c (hunk 1 above)
Comment (keith-zephyr) [reply tests/bluetooth/ctrl_isoal/src/main.c:401]: Can you add a comment describing why this call is needed?

Comment (keith-zephyr) [reply tests/bluetooth/ctrl_isoal/src/main.c:401]: Never mind - I see the comment at the prototype.

File: tests/bluetooth/ctrl_isoal/src/main.c [hunk 4]
```diff
@@ -336,46 +355,55 @@ static uint16_t add_framed_pdu_end(uint8_t *dataptr, uint8_t length, struct isoa
...
 {
 	debug_trace_func_call(__func__, "IN");
 
-	ztest_check_expected_value(sink_ctx);
-	ztest_check_expected_value(valid_pdu);
 	/* Return SDU buffer details as provided by the test */
-	ztest_copy_return_data(sdu_buffer, sizeof(*sdu_buffer));
-
-	return ztest_get_return_value();
+	zassert_not_null(sdu_buffer, NULL);
```
Comment (yperess) [inline tests/bluetooth/ctrl_isoal/src/main.c:365]: @keith-zephyr the assert is done here

File: tests/bluetooth/ctrl_isoal/src/main.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (yperess) [reply tests/bluetooth/ctrl_isoal/src/main.c:927]: Hard to parse that big file, I added a comment where the assert was added.

File: tests/bluetooth/ctrl_isoal/src/main.c (hunk 4 above)
Comment (keith-zephyr) [reply tests/bluetooth/ctrl_isoal/src/main.c:365]: Thanks

**Review by keith-zephyr — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
